flink-shaded-zookeeper-parent/flink-shaded-zookeeper-38/pom.xml

```diff
@@ -128,6 +131,7 @@ under the License.
                 <groupId>org.apache.zookeeper</groupId>
                 <artifactId>zookeeper</artifactId>
                 <version>${zookeeper.version}</version>
+                <optional>${flink.markBundledAsOptional}</optional>
             </dependency>
             <dependency>
                 <groupId>io.netty</groupId>
```

Comment:
we don't need that for BOMs?!

Reply:
That's not about BOMs
in general it does not make sense to add it for depedencies inside dependencyManagement
I removed it for others in dependencyManagement, thanks

Reply:
Can you elaborate a bit here? It doesn't make sense because dependencies from the `<dependencyManagement/>` section would need to be explicitly added to a module pom to make the dependency available and only then would we have to add the `<optional/>` tag?!

Looks like we have 4 locations in `flink-dist/pom.xml` and `flink-filesystems/flink-s3-fs-presto/pom.xml` where this also applies. :thinking:

Reply:
May be I was not clear enough, sorry
By saying 
>it does not make sense to add it for depedencies inside dependencyManagement

I ment not in general.
Here there is already existing dependency on zookeeper for every zookeeper module in dependencies which overrides the one from dependency management (optional tag).
Netty here comes only as a dependency from zookeeper and once zookeeper is marked  optional maven started to think that netty is also becoming optional
I don't think this is the case for poms you've mentioned